napari/components/_tests/test_layers_list.py

```diff
@@ -586,3 +587,11 @@ def test_ndim():
     # Remove layer
     layers.remove(layer_b)
     assert layers.ndim == 2
+
+
+def test_name_uniqueness():
+    v = ViewerModel()
+    v.add_image(np.random.random((10, 15)), name="Image [1]")
+    v.add_image(np.random.random((10, 15)), name="Image")
+    v.add_image(np.random.random((10, 15)), name="Image")
+    assert [x.name for x in v.layers] == ['Image [1]', 'Image', 'Image [2]']
```

Comment:
```suggestion
def test_name_uniqueness():
    layers = LayerList()
    layers.append(Image(np.random.random((10, 15)), name="Image [1]"))
    layers.append(Image(np.random.random((10, 15)), name="Image"))
    layers.append(Image(np.random.random((10, 15)), name="Image"))
    assert [x.name for x in layers] == ['Image [1]', 'Image', 'Image [2]']
```
Will this work? If so I'd prefer it as it keeps the `ViewerModel` import out of `test_layers_list.py` and is a little bit more minimal as there is no viewer there at all

Reply:
If not, then can you move test to `test_viewer_model.py`

Reply:
oh good point!  I think it _will_ work now, but not before this PR :)

Reply:
can we move this over too now https://github.com/napari/napari/blob/v0.4.2/napari/components/_tests/test_viewer_model.py#L446-L458

Reply:
or can leave there too, i don't mind it there. it's slightly duplicative (though not as advanced!) and maybe good to have test on viewer model too

Reply:
unfortunately, that one won't work yet until we let the layer know who it's parent is  (since the on_name event won't get hooked up)